File: Programmers/yerim/영어 끝말잇기.py
def solution(n, words):
    for i in range(1, len(words)):
        if (words[i][0] != words[i-1][-1]) or (words[i] in words[:i]) :
            # 탈락자 결정
            return [(i%n)+1, (i//n)+1]
    return [0, 0]

File: Programmers/hyeonyoung/영어 끝말잇기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <set>

using namespace std;

vector<int> solution(int n, vector<string> words)
{
    vector<int> answer = {0, 0};
    set<string> S;

    char c = words[0][0];
    int len = words.size();
    for (int i = 0; i < len; ++i)
    {
        if (c != words[i][0] || words[i].length() == 1)
        {
            answer = {i % n + 1, i / n + 1};
            break;
        }

        if (S.find(words[i]) == S.end())
        {
            S.insert(words[i]);
        }
        else
        {
            answer = {i % n + 1, i / n + 1};
            break;
        }

        c = words[i][words[i].length() - 1];
    }

    return answer;
}
